src/Entity/AI.ts

```diff
@@ -109,30 +110,42 @@ export class AI {
         this.game.entities.AIs.push(this);
     }
 
-    /** Finds the closest entity in a different team */
+    /* Finds the closest entity in a different team */
     public findTarget() {
+
         const rootPos = this.owner.rootParent.position.values;
         const team = this.owner.relations.values.team;
+
+        if (Entity.exists(this.target)) {
+
+            // If the AI already has a valid target within view distance, it's not necessary to find a new one
+
```

Comment:
Missing `team !== this.target.relations.values.team` check within this If statement.